### PR TITLE
fix: make consensus engine wait for a bit before giving up on worker

### PR DESCRIPTION
**Pull request by omerfirmak**
and if it happens despite the wait, let worker retry

### CODE REVIEW

**Review by colinlyguo — APPROVED** [review]

File: consensus/clique/clique.go [hunk 1]
```diff
@@ -648,6 +648,8 @@ func (c *Clique) Seal(chain consensus.ChainHeaderReader, block *types.Block, res
 	// Wait until sealing is terminated or delay timeout.
 	log.Trace("Waiting for slot to sign and propagate", "delay", common.PrettyDuration(delay))
 	go func() {
+		defer close(results)
```
Comment (georgehao) [inline consensus/clique/clique.go:651]: what will happen if don't close the results channel?

Comment (omerfirmak) [reply consensus/clique/clique.go:651]: [L656](https://github.com/scroll-tech/go-ethereum/pull/774/files#diff-1825330f8f75827945cb0144a6ab3dc034a3081f48ee96b0a318c5de9950832eR656) will block forever.

**Review by georgehao — APPROVED** [review]

File: consensus/clique/clique.go (hunk 1 above)
Comment (colinlyguo) [reply consensus/clique/clique.go:651]: btw, if the code goes the happy path (`case results <- block.WithSeal(header)`), would `close(results)` still be called? and if so, would it affect anything?

Comment (omerfirmak) [reply consensus/clique/clique.go:651]: Yeah, it would still be called but it doesn't change anything. No one ever interacts with resultCh again.

Comment (colinlyguo) [reply consensus/clique/clique.go:651]: got it.

**Pull request merged**